clients/roscpp/src/libros/topic_manager.cpp

```diff
@@ -656,6 +656,11 @@ bool TopicManager::requestTopic(const string &topic,
       int max_datagram_size = proto[4];
       int conn_id = connection_manager_->getNewConnectionID();
       TransportUDPPtr transport = connection_manager_->getUDPServerTransport()->createOutgoing(host, port, conn_id, max_datagram_size);
+      if (!transport)
+      {
+        ROSCPP_LOG_DEBUG("Error creating outgoing transport for [%s:%d]", host.c_str(), port);
+        return false;
+      }
```

Comment:
The DEBUG/return false seems to be convention in this method and can happen 4 other times in this else block. 

My expectation is that the code calling this will decide if a false return needs to be elevated to an error or if there are alternative strategies.